Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -176,33 +176,332 @@ struct HTTPHeader {
     let value: HTTPHeaderIndex
 }
 
+
+private extension UInt8 {
+    var isASCII: Bool {
+        return self <= 127
+    }
+
+    var maskedLowercase: UInt8 {
+        return self & 0xdf
+    }
+}
+
+
+/// Known HTTP header name. These should be created once and re-used as instances are memory intensive and "expensive" to create.
+public struct HTTPHeaderName {
+
+    /// Holds the bytes of the name + ': '
+    fileprivate let bytesWithSeperator: ContiguousArray<UInt8>
+    /// Holds the lowercased bases (masked) of the name.
+    fileprivate let lowerCaseBytes: ContiguousArray<UInt8>
+
+    /// Lowercase version of the name.
+    fileprivate let lowerCaseName: String
+
+    /// Original version of the name.
+    private let value: String
```

Comment:
Eh, I think throw it away, save ourselves the bytes and use a lower-case description.

Reply:
@Lukasa we can also recreate the String for the array... Let me do this for now (as this will also be inline with what the user used to construct it).